src/net/discovery/config.rs

```diff
@@ -0,0 +1,145 @@
+use crate::net::client::MpcHttpConnection;
+use crate::net::discovery::Error;
+use crate::net::discovery::PeerDiscovery;
+use axum::http::uri::{Authority, Scheme, Uri};
+use serde::{Deserialize, Deserializer};
+use std::fmt::Formatter;
+use std::path::PathBuf;
+
+/// Describes just the origin of a url, i.e.: "http\[s\]://\[authority\]", minus the path and
+/// query parameters
+#[derive(Debug, PartialEq, Eq)]
+struct Origin(Scheme, Authority);
+
+impl Origin {
+    fn to_uri(&self) -> Uri {
+        Uri::builder()
+            .scheme(self.0.clone())
+            .authority(self.1.clone())
+            .path_and_query("")
+            .build()
+            .unwrap()
+    }
+}
+
+impl<'de> Deserialize<'de> for Origin {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct UrlVisitor;
+        impl<'de> serde::de::Visitor<'de> for UrlVisitor {
+            type Value = Origin;
+
+            fn expecting(&self, formatter: &mut Formatter) -> std::fmt::Result {
+                formatter.write_str("a valid format origin made up of http[s]://<authority>")
+            }
+
+            fn visit_str<E: serde::de::Error>(self, v: &str) -> Result<Self::Value, E> {
+                let uri = v.parse::<Uri>().map_err(|err| E::custom(err.to_string()))?;
+                let parts = uri.into_parts();
+                let scheme = parts.scheme.ok_or_else(|| E::custom("missing scheme"))?;
+                let authority = parts
+                    .authority
+                    .ok_or_else(|| E::custom("missing authority"))?;
+                Ok(Origin(scheme, authority))
+            }
+        }
+        deserializer.deserialize_str(UrlVisitor)
+    }
+}
+
+/// Configuration values for a single peer helper of the MPC ring
+#[derive(Deserialize)]
+struct HelperConfig {
+    origin: Origin,
+}
+
+/// All config value necessary to discover all peer helpers of the MPC ring
+#[derive(Deserialize)]
+struct Config {
```

Comment:
this shouldn't be used anywhere except in tests so it should probably live in `test_fixture`?

In general I would expect each helper to start with `--role` flag that it can use to read its own part of the configuration file, so you don't really need to mix configuration for all helpers

Reply:
gotcha. for the version that exists here, it is true that all helpers would receive the same config file. We may have differences in the future, but I think this will be enough to get helpers talking to each other in a first draft.

hm, I was thinking it would be possible to use this in an intermediate length of time as a way to test servers across different machines. even in that scenario, you think we will just be executing a test?

Reply:
it should be fine to feed all helpers the same config file I think. they should only read the section that is relevant to them according to their assigned role. You would also need certificate information, private key that each helper should use to establish connection.

For instance, if helper got the role `H1` it should read the sections `H1.*` from config file

Reply:
gotcha, that makes sense. as we evolve this, it will be very easy to add more fields to the struct.

In the meantime, should we wait on martin to give the OK on using the `config` lib, or should I start implementing that now?

Reply:
yea my point here is that you can't really give this config to the helper initialization code. It is fine to keep it somewhere because you want to read the config file once, but helper server should only see one config (in this code I guess it would be `HelperConfig`).